### PR TITLE
Added obstacles, finished mapping extensions and rewrote how objects spawn and move to stay in sync.

### PR DESCRIPTION
**Pull request by caffeine239**
Added obstacles, finished mapping extensions and rewrote how objects spawn and move to stay in sync.

### CODE REVIEW

**Comment by isenmann** [comment]
Will have a look at the changes in the evening, but the video looks promising :-D

File: Assets/_Scripts/ObstacleHandling.cs [hunk 1]
```diff
@@ -1,23 +1,136 @@
...
+        }
+
+        _startPos.z += BeatsConstants.SWORD_OFFSET;
+        _midPos.z += BeatsConstants.SWORD_OFFSET;
+        _endPos.z += BeatsConstants.SWORD_OFFSET;
+    }
+
+    public void SetSize(float width, float height, float length)
+    {
+        GetChildByName(gameObject, "WallCore").transform.localScale = new Vector3(width, height, length);
```
Comment (isenmann) [inline Assets/_Scripts/ObstacleHandling.cs]: There is no child object with the name WallCore in the Wall prefab. Maybe you have forgot to commit the change to the prefab there?

File: Assets/_Scripts/ObstacleHandling.cs [hunk 2]
```diff
@@ -1,23 +1,136 @@
...
+
+        SetPosition(width, height, length);
+    }
+
+    public void SetPosition(float width, float height, float length)
     {
-        
+        GetChildByName(gameObject, "WallCore").transform.localPosition = new Vector3(0f, height * 0.5f, length * 0.5f);
```
Comment (isenmann) [inline Assets/_Scripts/ObstacleHandling.cs]: Same applies to here, no WallCore child object in the Wall prefab

**Comment by caffeine239** [comment]
I did. Will upload when I'm home.

**Comment by caffeine239** [comment]
Done.

**Pull request merged**